rust/rope/src/subset.rs

```diff
@@ -48,57 +48,55 @@ impl SubsetBuilder {
 
     pub fn build(mut self) -> Subset {
         if self.e > self.b {
-            self.dels.push((self.b, self.e));
+            self.ranges.push((self.b, self.e));
         }
-        Subset(self.dels)
+        Subset(self.ranges)
     }
 }
 
 impl Subset {
-    // mostly for testing
-    pub fn apply_to_string(&self, s: &str) -> String {
+    /// Mostly for testing.
+    pub fn delete_from_string(&self, s: &str) -> String {
         let mut result = String::new();
-        for (b, e) in self.range_iter(s.len()) {
+        for (b, e) in self.complement_iter(s.len()) {
             result.push_str(&s[b..e]);
         }
         result
     }
 
     // Maybe Subset should be a pure data structure and this method should
     // be a method of Node.
-    pub fn apply<N: NodeInfo>(&self, s: &Node<N>) -> Node<N> {
+    /// Builds a version of `s` with all the elements in this `Subset` deleted from it.
+    pub fn delete_from<N: NodeInfo>(&self, s: &Node<N>) -> Node<N> {
         let mut b = TreeBuilder::new();
-        for (beg, end) in self.range_iter(s.len()) {
+        for (beg, end) in self.complement_iter(s.len()) {
             s.push_subseq(&mut b, Interval::new_closed_open(beg, end));
         }
         b.build()
     }
 
-    /// The length of the resulting sequence.
+    /// The length of the resulting sequence after deleting this subset.
     ///
-    /// `self.apply_to_string(s).len() = self.len(s.len())`
-    pub fn len(&self, base_len: usize) -> usize {
+    /// `self.delete_from_string(s).len() = self.len(s.len())`
+    pub fn len_after_delete(&self, base_len: usize) -> usize {
         self.0.iter().fold(base_len, |acc, &(b, e)| acc - (e - b))
     }
 
-    /// Determine whether the subset is trivial, ie applying it does not
-    /// change the sequence. In purely set theoretic terms, this is the
-    /// same as testing equality with the universal set.
-    pub fn is_trivial(&self) -> bool {
+    /// Determine whether the subset is empty.
+    /// In this case deleting it would do nothing.
+    pub fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
 
     #[doc(hidden)]
     // Access to internal state, shouldn't really be part of public API
-    // Perhaps exposing an iterator over deleted regions would be more suitable,
-    // but it's more of a hassle.
-    pub fn _deletions(&self) -> &[(usize, usize)] {
+    pub fn _ranges(&self) -> &[(usize, usize)] {
         &self.0
     }
 
-    /// Compute the intersection of two subsets. In other words, an element exists in the
-    /// resulting subset iff it exists in both inputs.
-    pub fn intersect(&self, other: &Subset) -> Subset {
+    /// Compute the union of two subsets. In other words, an element exists in the
+    /// resulting subset if it exists either input.
```

Comment:
Oh that was just an accidental omission when I edited it to describe union instead of intersection. I just improved it.